optuna/_callbacks.py

```diff
@@ -91,10 +91,16 @@ class RetryFailedTrialCallback:
             The max number of times a trial can be retried. Must be set to :obj:`None` or an
             integer. If set to the default value of :obj:`None` will retry indefinitely.
             If set to an integer, will only retry that many times.
+        inherit_intermediate:
```

Comment:
As far as I know, optuna code calls `intermediate_values` without an abbreviation (e.g., [plot_intermediate_values]( https://github.com/optuna/optuna/blob/master/optuna/visualization/_intermediate_values.py)). It will be a bit long argument name, but could you rename it `inherit_intermediate_values`?

```suggestion
        inherit_intermediate_values:
```